src/psyclone/gocean1p0.py

```diff
@@ -1304,3 +1361,30 @@ def nargs(self):
     def index_offset(self):
         ''' Return the grid index-offset that this kernel expects '''
         return self._index_offset
+
+
+class GOACCDataDirective(ACCDataDirective):
+    '''
+    Sub-classes ACCDataDirective to provide an API-specific implementation
+    of data_on_device().
+
+    '''
+    def data_on_device(self, parent):
+        '''
+        Adds nodes into a Schedule to flag that each of the objects required
+        by the kernels in the data region is now on the device. We do
+        this by setting the data_on_device attribute to .true.
+
+        :param parent: the node in the Schedule to which to add nodes
```

Comment:
It's not a node in the schedule, it's a node in the f2pygen fortran representation isn't it?

Reply:
Oops, yes. Got my ASTs confused. I've clarified the doc string and corrected the type description.